engine/src/gradient.cpp

```diff
@@ -665,15 +665,16 @@ IO_stat MCGradientFillSerialize(MCGradientFill *p_gradient, MCObjectOutputStream
 		t_stat = p_stream . WriteU8(p_gradient -> repeat);
 	if (t_stat == IO_NORMAL)
 		t_stat = p_stream . WriteU8(p_gradient -> ramp_length);
-	
-	MCPoint *t_point;
-	t_point = &p_gradient -> origin;
-	for(int i = 0; i < 3 && t_stat == IO_NORMAL; ++i)
-	{
-		t_stat = p_stream . WriteS16(t_point[i] . x);
-		if (t_stat == IO_NORMAL)
-			t_stat = p_stream . WriteS16(t_point[i] . y);
-	}
+
+    auto t_write_point = [&p_stream, &t_stat](MCPoint& p) {
```

Comment:
Pragmatically the number of things that might be written to an MCObjectStream is quite small - so adding a WriteMCPoint to MCObjectStrem seems perfectly sensible, to me at least :)